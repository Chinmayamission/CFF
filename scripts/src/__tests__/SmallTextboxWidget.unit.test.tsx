import React from "react";
import { shallow, mount, render } from 'enzyme';
import CustomForm from "src/form/CustomForm";

it('renders small text box widget correctly', () => {
  let schema = {
    "type": "object",
    "title": "Form",
    "properties": {
      "interests_other": {
        "type": "string"
      }
    }
  };
  let uiSchema = {
    "interests_other": {
      "ui:widget": "cff:smallTextbox"
    }
  }
  const wrapper = render(
    <CustomForm schema={schema} uiSchema={uiSchema} />
  );
  // expect(wrapper.exists(".root_interests")).toEqual(true);
  expect(wrapper).toMatchSnapshot();
}); 