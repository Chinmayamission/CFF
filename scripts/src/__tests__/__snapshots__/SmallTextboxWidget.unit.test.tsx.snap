// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders small text box widget correctly 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    autocomplete="false"
    class="rjsf"
  >
    <div
      class="form-group field field-object"
    >
      <div
        class="container-fluid"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Form
        </h2>
        <div />
        <div
          class="row"
        >
          <div
            class="form-group field field-string  col"
          >
            <label
              class="control-label"
              for="root_interests_other"
            >
              interests_other
            </label>
            <input
              class=""
              id="root_interests_other"
              label="interests_other"
              placeholder=""
              type="text"
              value=""
            />
            <div />
            <div />
          </div>
        </div>
      </div>
      <div />
      <div />
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;
